src/util/simplify_expr.cpp

```diff
@@ -1777,9 +1777,26 @@ bool simplify_exprt::simplify_byte_extract(byte_extract_exprt &expr)
   const auto bits=
     expr2bits(expr.op(), expr.id()==ID_byte_extract_little_endian);
 
-  // exact match of length only - otherwise we might lose bits of
-  // flexible array members at the end of a struct
-  if(bits.has_value() && mp_integer(bits->size()) == *el_size + *offset * 8)
+  // make sure we don't lose bits with structs containing flexible array members
+  const bool struct_has_flexible_array_member = has_subtype(
```

Comment:
⛏ Might be worth factoring that out into a utility to avoid someone recoding that at the next occasion...

Reply:
I'll in this not follow the advice for two reasons: 1) This is an over-approximating check, even `struct_has_flexible_array_member == true` need not mean that there genuinely are flexible array members (which is ok in this context, it just mean we don't simplify although we possibly could); 2) I haven't found any other bits of the code outside the C type checker that currently try to deal with this special case. Should a need for that arise, of course, we should refactor. I hope that's ok!